R/lag.R

```diff
@@ -91,6 +91,10 @@ step_lag_new <-
 
 #' @export
 prep.step_lag <- function(x, training, info = NULL, ...) {
+  if (!all(x$lag == as.integer(x$lag))) {
+    rlang::abort("step_lag requires 'lag' argument to be integer valued.")
```

Comment:
```suggestion
    rlang::abort("step_lag() requires 'lag' argument to be integer-valued.")
```

Reply:
I'm on board for moving this from `bake()` to `prep()`. Could we document in this PR (here is fine) why that is?

Reply:
There are 3 places where we can put `abort()` in a step. `step_lag()`, `prep.step_lag()` and `bake.step_lag()`.

If we put in in `step_lag()` you won't be able to run the following code

```r
library(recipes)
recipe(~., data = mtcars) |>
  step_lag(mpg, lag = 0.5)
```

If you put it in `prep.step_lag()`, we get the error once we call `prep()` or `fit()` if it were in a workflow.

And if it is put in `bake.step_lag()`, then we get the error every time we try to `bake()` or `predict()` with it. 

I don't like the first option, because it means that you can't create a wrong specification. And it makes testing harder.

I don't like the last option, because then code would run every time we try to apply it to new data. Which is redundant since the input isn't changed from time to time. Hence why I put it in `prep.step_lag()`